src/main/scala/com/actian/spark_vector/vector/VectorJDBC.scala

```diff
@@ -15,12 +15,49 @@
  */
 package com.actian.spark_vector.vector
 
-import java.sql._
+import java.sql.{ Connection, DriverManager, PreparedStatement, ResultSet, Statement }
 import java.util.Properties
-import com.actian.spark_vector.util.ResourceUtil
-import com.actian.spark_vector.vector.ErrorCodes._
+
 import org.apache.spark.Logging
-import com.actian.spark_vector.sql.VectorRelation
+import org.apache.spark.sql.Row
+
+import com.actian.spark_vector.Profiling
+import com.actian.spark_vector.util.ResourceUtil.RichExtractableManagedResource
+import com.actian.spark_vector.vector.ErrorCodes.{ InvalidDataType, NoSuchTable, SqlException, SqlExecutionError }
+
+import resource.managed
+
+/** Iterator over an ResultSet */
+abstract class ResultSetIterator[T](result: ResultSet) extends Iterator[T] {
+  def extractor: ResultSet => T
```

Comment:
you could make this protected as I think it is used only along the chain of class inheritance